hacep/src/main/java/org/kie/hacep/consumer/CommandHandler.java

```diff
@@ -74,9 +78,14 @@ public CommandHandler(KieSessionContext kieSessionContext,
     @Override
     public void visit(FireAllRulesCommand command) {
         int fires = kieSessionContext.getKieSession().fireAllRules();
-        producer.produceSync(envConfig.getKieSessionInfosTopicName(),
-                             command.getId(),
-                             fires);
+        FireAllRuleMessage msg = new FireAllRuleMessage(command.getId(), fires);
+
+        // command.isPermittedForReplicas() is true but only Leader should produce a message
+        if (DroolsExecutor.getInstance().isLeader()) {
```

Comment:
I don't like to use this workaround to get current status but it is the simplest way I found